src/posix/platform/infra_if.cpp

```diff
@@ -277,7 +289,43 @@ bool InfraNetif::IsRunning(void) const
 
     close(sock);
 
-    return (ifReq.ifr_flags & IFF_RUNNING) && HasLinkLocalAddress();
+    return ifReq.ifr_flags;
+}
+
+void InfraNetif::CountAddresses(uint32_t &aLinkLocalAddresses,
+                                uint32_t &aUniqueLocalAddresses,
+                                uint32_t &aGlobalUnicastAddresses) const
+{
+    struct ifaddrs *ifAddrs = nullptr;
+
+    aLinkLocalAddresses     = 0;
+    aUniqueLocalAddresses   = 0;
+    aGlobalUnicastAddresses = 0;
+
+    if (getifaddrs(&ifAddrs) < 0)
+    {
+        otLogCritPlat("failed to get netif addresses: %s", strerror(errno));
+        DieNow(OT_EXIT_ERROR_ERRNO);
```

Comment:
It might not be expected for the TBR to die when it fails? e.g. when the AP is off temporarily, the TBR STA can still work as a Thread Router.
And in general, I don't think the behavior of collecting metrics will cause the exit of the process

Reply:
This line is inherited from the previous implementation of `IsRunning`. I'm not sure why we decided to kill `otbr-agent` in such a situation.

Reply:
agree with @librasungirl  this is not a critical error.